### PR TITLE
RT-577 :6mo button stays selected after zooming

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview
We use the onZoomStart on that, we ate removing the focus of the range selector.
Changes in test cases because we rename the component function name and MatButtons Id.

## How it was tested
Tested using run showcase app locally.

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [ ] I have included screen shots for changes that affect the user interface
- [x] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

**Comment by yashpmohod** [comment]
we have created another PR for the RT-577 branch.

**Review by semicolin — CHANGES_REQUESTED** [review]
I just noticed that the unit tests are using the real FhirChartConfigurationService. We should probably be mocking that. You can add a test with mock `configService.timelineRange$` to fill the gap in  test coverage.

**Comment by yashpmohod** [comment]
Working on this

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts [hunk 1]
```diff
@@ -181,8 +193,17 @@ describe('RangeSelectorComponent', () => {
     const componentMindate = new Date();
     const monthCount = 1;
     componentMindate.setMonth(componentMindate.getMonth() - monthCount);
-    let months = component.calculateMonthDiff(componentMindate, componentMaxdate)
+    const months = component.calculateMonthDiff(componentMindate, componentMaxdate)
     expect(months).toEqual(monthCount)
   })
 
+  it('should subscribe timelineRange when the component initializes and set minDate and maxDate', async () => {
```
[the remark this answers is not in the document]
Comment (semicolin) [reply projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts:200]: actually, if you apply my suggested changes below, you don't need either

**Review by semicolin — CHANGES_REQUESTED** [review]

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts:196]: Okay, I will apply your suggested chnages.

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts [hunk 2]
```diff
@@ -181,8 +193,17 @@ describe('RangeSelectorComponent', () => {
...
+    mockConfigServiceManager.timelineRange$.subscribe((result: any) => {
+      expect(component.minDate).toEqual(new Date(result.min))
+      expect(component.maxDate).toEqual(new Date(result.max))
+      const months = component.calculateMonthDiff(new Date(result.min), new Date(result.max))
+      expect(component.selectedButton).toEqual(months || true)
+    })
```
[the remark this answers is not in the document]
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts:202]: Okay I will change test case as per your suggestion.

**Review by semicolin — APPROVED** [review]

**Pull request merged**